packages/xforms-engine/test/parse/xpath/semantic-analysis.test.ts

```diff
@@ -0,0 +1,91 @@
+import { expressionParser } from '@getodk/xpath/expressionParser.js';
+import { describe, expect, it } from 'vitest';
+import {
+	findLocationPathSubExpressionNodes,
+	isTranslationExpression,
+} from '../../../src/parse/xpath/semantic-analysis.ts';
+
+describe('Semantic analysis', () => {
+	describe('FunctionCall', () => {
+		describe('translation', () => {
+			interface IsTranslationExpressionCase {
+				readonly description: string;
+				readonly expression: string;
+				readonly expected: boolean;
+			}
+
+			it.each<IsTranslationExpressionCase>([
+				{
+					description: 'jr:itext call referencing string Literal id',
+					expression: 'jr:itext("id")',
+					expected: true,
+				},
+				{
+					description: 'jr:itext call referencing RelativeLocationPath id',
+					expression: 'jr:itext(itemset-item-label-reference)',
+					expected: true,
+				},
+				{
+					description: 'jr:itext call referencing AbsoluteLocationPath id',
+					expression: 'jr:itext(/some/absolute/reference)',
+					expected: true,
+				},
+				{
+					description: 'unexpected Number argument',
+					expression: 'jr:itext(1)',
+					expected: false,
+				},
+				{
+					description: 'FuntionName references another function',
+					expression: 'some-other-fn()',
+					expected: false,
+				},
+				{
+					description: 'itext is substring of FunctionName, not full name',
+					expression: 'some-fn-with-itext-in-name()',
+					expected: false,
+				},
+
+				{
+					description:
+						'we do not currently check prefixes for function call analysis, typical usage has common namespace mapping',
```

Comment:
This seems ok.

Reply:
Yeah... it's not my favorite thing, but I really wouldn't want to connect up form-derived evaluation context to static analysis unless/until we have a compelling reason, like a bug in the wild.